app/tray/Account/Requests/TransactionRequest/TxMainNew/overview.js

```diff
@@ -60,6 +51,31 @@ const DeployContractOverview = () => <div>Deploying Contract</div>
 const GenericContractOverview = ({ method }) => <div>{`Calling Contract Method ${method}`}</div>
 const DataOverview = () => <div>Sending data</div>
 
+const actionOverviews = {
+  'erc20:transfer': SendOverview,
+  'erc20:approve': ApproveOverview,
+  ens: EnsOverview
+}
+
+const renderActionOverview = (action, index) => {
+  const { id = '', data } = action
+  const key = id + index
+  const [, actionType] = id.split(':')
```

Comment:
the idea here was to separate out the action class and action type so that we can have components that handle all the actions for one class, ie erc20 stuff. this pattern works for now but lets at least keep the concept of the action class so that as this grows we can be flexible

Reply:
yeah fair enough. can we just keep the variable there? I think in general that gives visibility in the codebase as opposed to the empty comma syntax. otherwise this all looks good